java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/ElasticBulkProcessorAdapter.java

```diff
@@ -114,34 +74,94 @@ public void awaitClose(long timeout, TimeUnit unit) throws InterruptedException
     }
 
     @Override
-    public void addRow(SinkRow row, String indexName, RequestTracker requestTracker) {
-        final String index = (String) row.get(0);
-        final String key = (String) row.get(1);
-        String doc = (String) row.get(2);
-
+    public void addRow(String index, String key, String doc) {
         UpdateRequest updateRequest;
-        if (indexName != null) {
-            updateRequest = new UpdateRequest(indexName, "_doc", key).doc(doc, XContentType.JSON);
-        } else {
-            updateRequest = new UpdateRequest(indexName, "_doc", key).doc(doc, XContentType.JSON);
-        }
+        updateRequest = new UpdateRequest(index, "_doc", key).doc(doc, XContentType.JSON);
         updateRequest.docAsUpsert(true);
-        requestTracker.addWriteTask();
+        this.requestTracker.addWriteTask();
         this.esBulkProcessor.add(updateRequest);
     }
 
     @Override
-    public void deleteRow(SinkRow row, String indexName, RequestTracker requestTracker) {
-        final String index = (String) row.get(0);
-        final String key = (String) row.get(1);
-
+    public void deleteRow(String index, String key) {
         DeleteRequest deleteRequest;
-        if (indexName != null) {
-            deleteRequest = new DeleteRequest(indexName, "_doc", key);
+        deleteRequest = new DeleteRequest(index, "_doc", key);
+        this.requestTracker.addWriteTask();
+        this.esBulkProcessor.add(deleteRequest);
+    }
+}
+
+class BulkListener
```

Comment:
Shall we move the `BulkListener` to a separate file, since it's shared by the two implementations.